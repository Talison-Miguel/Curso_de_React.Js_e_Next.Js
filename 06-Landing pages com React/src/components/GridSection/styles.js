import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const Grid = styled.div`
    ${({ theme }) => css``}
`;

export const GridElement = styled.div`
    ${({ theme }) => css``}
`;
