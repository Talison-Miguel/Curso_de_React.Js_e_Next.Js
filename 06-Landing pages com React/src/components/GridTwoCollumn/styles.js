import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const TextContainer = styled.div`
    ${({ theme }) => css``}
`;

export const ImageContainer = styled.div`
    ${({ theme }) => css``}
`;
