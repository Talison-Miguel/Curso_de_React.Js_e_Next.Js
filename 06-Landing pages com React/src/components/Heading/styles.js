import styled, { css } from 'styled-components';

const titleSize = {
    small: (theme) => css`
        font-size: ${theme.font.sizes.medium};
    `,
    medium: (theme) => css`
        font-size: ${theme.font.sizes.large};
    `,
    big: (theme) => css`
        font-size: ${theme.font.sizes.xlarge};
    `,
    huge: (theme) => css`
        font-size: ${theme.font.sizes.xhuge};
        ${mediaFont(theme)}
    `,
};

const mediaFont = (theme) => css`
    @media ${theme.media.lessThanMedium} {
        font-size: ${theme.font.sizes.xlarge};
    }
`;

const titleCase = (uppercase) => css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
    ${({ theme, colordark, size, uppercase }) => css`
        color: ${colordark ? theme.colors.primaryColor : theme.colors.white};
        ${titleSize[size](theme)}
        ${titleCase(uppercase)}
    `}
`;
