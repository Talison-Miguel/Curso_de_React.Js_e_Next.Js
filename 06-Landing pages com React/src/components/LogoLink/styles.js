import styled, { css } from 'styled-components';

export const Container = styled.a`
    ${({ theme }) => css`
        text-decoration: none;
        color: inherit;
        display: flex;
        align-items: center;

        > img {
            max-height: 3rem;
        }
    `}
`;
