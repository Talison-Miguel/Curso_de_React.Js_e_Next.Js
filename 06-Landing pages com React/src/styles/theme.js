export const theme = {
    colors: {
        primaryColor: '#0a1128',
        secondaryBg: '#dc143c',
        white: '#fff',
    },
    font: {
        family: {
            default: "'Open Sans', sans-serif",
            secondary: "'Montserrat', sans-serif",
        },
        sizes: {
            xmall: '8rem',
            small: '1.6rem',
            medium: '2.4rem',
            large: '3.2rem',
            xlarge: '4.0rem',
            xxlarge: '4.8rem',
            huge: '5.6rem',
            xhuge: '6.4rem',
        },
    },
    media: {
        lessThanMedium: '(max-width: 768px)',
    },
    spacings: {
        xmall: '8rem',
        small: '1.6rem',
        medium: '2.4rem',
        large: '3.2rem',
        xlarge: '4.0rem',
        xxlarge: '4.8rem',
        huge: '5.6rem',
        xhuge: '6.4rem',
    },
};
