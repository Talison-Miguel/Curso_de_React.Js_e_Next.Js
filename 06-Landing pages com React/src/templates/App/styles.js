import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    ${({ theme }) => css`
        background-color: ${theme.colors.secondaryBg};
    `}
`;
