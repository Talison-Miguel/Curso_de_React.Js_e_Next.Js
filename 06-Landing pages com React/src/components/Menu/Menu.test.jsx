import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

import { Menu } from '.';
import linksMock from '../NavLinks/mock';
import { theme } from '../../styles/theme';

const logoData = {
    text: 'Logo',
    link: '#target',
};

describe('<Menu />', () => {
    it('Should render logo and main menu nav', () => {
        const { container } = renderTheme(<Menu links={linksMock} logoData={logoData} />);
        expect(screen.getByRole('heading', { name: 'Logo' })).toBeInTheDocument();

        expect(screen.getByRole('navigation', { name: 'Main menu' })).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    it('Should render menu mobile and button for open and close the menu', () => {
        const { container } = renderTheme(<Menu links={linksMock} logoData={logoData} />);

        const button = screen.getByLabelText('Open/Close menu');
        const menuContainer = button.nextSibling;

        expect(button).toHaveStyleRule('display', 'none');

        expect(screen.getByLabelText('Open menu')).toBeInTheDocument();

        fireEvent.click(button);
    });

    it('Should not render links', () => {
        const { container } = renderTheme(<Menu logoData={logoData} />);

        expect(screen.queryByRole('navigation', { name: 'Main menu' }).firstChild).not.toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
