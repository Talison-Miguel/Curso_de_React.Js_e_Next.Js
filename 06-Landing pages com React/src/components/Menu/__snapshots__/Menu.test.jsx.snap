// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Should not render links 1`] = `
.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  border-bottom: #DDDDDD;
  background: #fff;
  transition: all 300ms ease-in-out;
}

.c2 >.c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c2 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0a1128;
  color: #fff;
  border: none;
  display: none;
  pointer-events: none;
}

.c0 >svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c7 {
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;
}

.c7 >img {
  max-height: 3rem;
}

.c6 {
  color: #0a1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c8 {
  display: flex;
  flex-flow: row wrap;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width: 768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
    visibility: visible;
    opacity: 1;
  }

  .c2 >.c3 {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    align-items: center;
    overflow-y: auto;
  }

  .c2 {
    padding-bottom: 3.2rem;
    display: flex;
    justify-content: center;
  }
}

@media (max-width: 768px) {
  .c5 {
    display: block;
    text-align: center;
    padding: 4.8rem 0;
  }
}

@media (max-width: 768px) {
  .c0 {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}

@media (max-width: 768px) {
  .c8 {
    flex-flow: column wrap;
    align-content: center;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
    visible="false"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
    visible="false"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6"
          colordark="true"
          uppercase="true"
        >
          <a
            class="c7"
            href="#target"
          >
            <span>
              Logo
            </span>
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c8"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Menu /> Should render logo and main menu nav 1`] = `
.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  border-bottom: #DDDDDD;
  background: #fff;
  transition: all 300ms ease-in-out;
}

.c2 >.c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c2 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0a1128;
  color: #fff;
  border: none;
  display: none;
  pointer-events: none;
}

.c0 >svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c7 {
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;
}

.c7 >img {
  max-height: 3rem;
}

.c6 {
  color: #0a1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c8 {
  display: flex;
  flex-flow: row wrap;
}

.c9 {
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c9::after {
  content: '';
  position: absolute;
  bottom: 0.76rem;
  left: 50%;
  width: 0%;
  height: 0.2rem;
  background: #dc143c;
  transition: all 300ms ease-in-out;
}

.c9:hover::after {
  left: 25%;
  width: 50%;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width: 768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
    visibility: visible;
    opacity: 1;
  }

  .c2 >.c3 {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    align-items: center;
    overflow-y: auto;
  }

  .c2 {
    padding-bottom: 3.2rem;
    display: flex;
    justify-content: center;
  }
}

@media (max-width: 768px) {
  .c5 {
    display: block;
    text-align: center;
    padding: 4.8rem 0;
  }
}

@media (max-width: 768px) {
  .c0 {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}

@media (max-width: 768px) {
  .c8 {
    flex-flow: column wrap;
    align-content: center;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
    visible="false"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
    visible="false"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6"
          colordark="true"
          uppercase="true"
        >
          <a
            class="c7"
            href="#target"
          >
            <span>
              Logo
            </span>
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c8"
        >
          <a
            class="c9"
            href="#target1"
            target="_self"
          >
            Link 1
          </a>
          <a
            class="c9"
            href="#target2"
            target="_self"
          >
            Link 2
          </a>
          <a
            class="c9"
            href="#target3"
            target="_self"
          >
            Link 3
          </a>
          <a
            class="c9"
            href="#target4"
            target="_self"
          >
            Link 4
          </a>
          <a
            class="c9"
            href="#target5"
            target="_self"
          >
            Link 5
          </a>
          <a
            class="c9"
            href="#target6"
            target="_self"
          >
            Link 6
          </a>
          <a
            class="c9"
            href="#target7"
            target="_self"
          >
            Link 7
          </a>
          <a
            class="c9"
            href="#target8"
            target="_self"
          >
            Link 8
          </a>
          <a
            class="c9"
            href="#target9"
            target="_self"
          >
            Link 9
          </a>
          <a
            class="c9"
            href="#target10"
            target="_self"
          >
            Link 10
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
